metricbeat/module/vsphere/resourcepool/resourcepool_test.go

```diff
@@ -0,0 +1,131 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package resourcepool
+
+import (
+	"testing"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/govmomi/simulator"
+)
+
+func TestFetchEventContents(t *testing.T) {
+	model := simulator.VPX()
+	if err := model.Create(); err != nil {
+		t.Fatal(err)
+	}
+
+	ts := model.Service.NewServer()
+	defer ts.Close()
+
+	f := mbtest.NewReportingMetricSetV2WithContext(t, getConfig(ts))
+	events, errs := mbtest.ReportingFetchV2WithContext(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	}
```

Comment:
I wonder if it's worth it to log each error individually like:

```
    for _, err := range errs {
        t.Logf("Error: %v\n", err)
    }
```

WDYT @kush-elastic?

Reply:
There's a way to do it in one step:

```go
package main

import (
	"errors"
	"fmt"
)

func main() {
	err := errors.New("one")
	err2 := errors.New("two")
	err3 := errors.New("three")
	fmt.Println(errors.Join(err, nil, err2, err3))
}
```

https://go.dev/play/p/P8AUwo3WXUf